src/main/java/world/bentobox/bentobox/listeners/JoinLeaveListener.java

```diff
@@ -205,8 +205,8 @@ private void updateIslandRange(User user) {
             if (island != null) {
                 // Check if new owner has a different range permission than the island size
                 int range = user.getPermissionValue(plugin.getIWM().getAddon(island.getWorld()).map(GameModeAddon::getPermissionPrefix).orElse("") + "island.range", island.getProtectionRange());
-                // Range cannot be greater than the island distance
-                range = Math.min(range, plugin.getIWM().getIslandDistance(island.getWorld()));
+                // Range cannot be greater than the island distance * 2
+                range = Math.min(range, 2 * plugin.getIWM().getIslandDistance(island.getWorld()));
```

Comment:
I am not sure if this is a correct way how to do it.

I would prefer to calculate by offset from the actual island center.

Reply:
The protection range is now relative to the protection center location, so if the player is to be given the ability to use all of their island area, then their protection range must be able to go up to 2x island distance away from that point.

For example:

Island range = 400
Island center = 0,0
Island location = 399, 0
Island protection range = 800

This last value will enable them to operate anywhere with all of the island space.